archive/extract_test.go

```diff
@@ -167,3 +178,12 @@ func testExtract(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 }
+
+func testPathPerms(t *testing.T, parentDir, path string, expectedMode os.FileMode) {
```

Comment:
Since we're calling this function only in one use case, maybe we can put it inside the for loop. I'm not sure what will be clearer for the user.

Reply:
Aye, I called this function in two places before and forgot to change it back. I would update but I'm having problems with my IDE... let's just leave it out of laziness.